### PR TITLE
configure: fix for FIPS out-of-tree builds

### PR DESCRIPTION
**Pull request by elms**
Check for fips files relative to source directory.
Simplify validation logic.

### CODE REVIEW

**Review by douzzer — APPROVED** [review]
tested out-of-tree FIPS build, and it now works.  thanks Jeff!

File: configure.ac [hunk 1]
```diff
@@ -165,43 +165,45 @@ AC_ARG_ENABLE([fips],
...
     ],
-    ["no"],[FIPS_VERSION="none"],
+    [yes],
     [
         # FIPS v1
         ENABLED_FIPS="yes"
         FIPS_VERSION="v1"
+    ],
+    [
+        AC_MSG_ERROR([Invalid value for --enable-fips $ENABLED_FIPS (allowed: ready, rand, v1, v2)])
```
Comment (douzzer) [inline configure.ac:188]: should be `\"$ENABLED_FIPS\"` here.

**Review by douzzer — CHANGES_REQUESTED** [review]
good idea to have `--enable-fips` error out on nonsense values.

File: configure.ac (hunk 1 above)
Comment (elms) [reply configure.ac:188]: Happy to add that. Is that for clarity or did you actually have an issue when running?

Comment (douzzer) [reply configure.ac:188]: just clarity.

Comment (elms) [reply configure.ac]: Done

**Review by douzzer — APPROVED** [review]
LGTM

**Pull request merged**